pkg/apis/syn/v1alpha1/tenant_types_test.go

```diff
@@ -82,3 +84,211 @@ func TestTenant_ApplyTemplate(t *testing.T) {
 		})
 	}
 }
+
+func TestTenant_CreateServiceAccount(t *testing.T) {
+	tenant := &synv1alpha1.Tenant{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "my-tenant",
+			Namespace: "my-namespace",
+		},
+	}
+
+	sa := tenant.NewServiceAccount()
+
+	assert.Equal(t, tenant.Name, sa.Name)
+	assert.Equal(t, tenant.Namespace, sa.Namespace)
+	assertManagedByLabel(t, sa.ObjectMeta, "lieutenant")
+	// TODO Assert owner reference
+}
+
+func TestTenant_CreateRole(t *testing.T) {
+	tenant := &synv1alpha1.Tenant{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "my-tenant",
+			Namespace: "my-namespace",
+		},
+	}
+
+	role := tenant.NewRole()
+
+	assert.Equal(t, tenant.Name, role.Name)
+	assert.Equal(t, tenant.Namespace, role.Namespace)
+	assertManagedByLabel(t, role.ObjectMeta, "lieutenant")
+	// TODO Assert owner reference
+	require.Len(t, role.Rules, 1)
+
+	assert.Equal(t, []string{synv1alpha1.SchemeGroupVersion.Group}, role.Rules[0].APIGroups)
+	assert.Len(t, role.Rules[0].Resources, 2)
+	assert.Contains(t, role.Rules[0].Resources, "tenants")
+	assert.Contains(t, role.Rules[0].Resources, "clusters")
+	assert.Len(t, role.Rules[0].Verbs, 1)
+	assert.Contains(t, role.Rules[0].Verbs, "get")
+}
+
+func TestTenant_CreateRoleBinding(t *testing.T) {
```

Comment:
it probably got forgotten
```suggestion
func TestTenant_NewRoleBinding(t *testing.T) {
```